runtime-modules/referendum/src/lib.rs

```diff
@@ -646,16 +645,11 @@ impl<T: Trait<I>, I: Instance> Mutations<T, I> {
     fn vote(
         account_id: &<T as frame_system::Trait>::AccountId,
         commitment: &T::Hash,
-        stake: &Balance<T, I>,
+        stake: &BalanceOf<T>,
         current_cycle_id: &u64,
     ) -> Result<(), Error<T, I>> {
-        // lock stake amount
-        T::Currency::set_lock(
-            T::LockId::get(),
-            account_id,
-            *stake,
-            WithdrawReason::Transfer.into(),
-        );
+        // Should call after `can_vote`
+        T::StakingHandler::lock(account_id, *stake);
```

Comment:
WithdrawReasons are incorrect. The latest version of the staking handler should contain suitable method to set a proper withdraw reason.